### PR TITLE
Convert BTLocalPaymentResult to Swift

### PR DESCRIPTION
**Pull request by scannillo**
### Summary of changes

JIRA ticket DTBTSDK-2276

- Convert `BTLocalPaymentResult` to Swift
- Convert `BTPaymentFlowResult` to Swift
- Note:
   - SPM & CocoaPod builds expected to fail until `PaymentFlow` is fully converted

### Checklist

- ~Added a changelog entry~

### Authors
@scannillo

### CODE REVIEW

File: Sources/BraintreePaymentFlow/BTLocalPaymentResult.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
+@objcMembers public class BTLocalPaymentResult: BTPaymentFlowResult {
+    
+    /// The billing address.
+    public let billingAddress: BTPostalAddress
```
Comment (KunJeongPark) [inline Sources/BraintreePaymentFlow/BTLocalPaymentResult.swift:4]: Are all these fields non-optional?

Comment (scannillo) [reply Sources/BraintreePaymentFlow/BTLocalPaymentResult.swift:4]: These are non-optional, yes. `BTLocalPaymentResult` is instantiated internally (see TODO below in file).

File: Sources/BraintreePaymentFlow/BTLocalPaymentResult.swift [hunk 2]
```diff
@@ -0,0 +1,58 @@
+@objcMembers public class BTLocalPaymentResult: BTPaymentFlowResult {
```
Comment (KunJeongPark) [inline Sources/BraintreePaymentFlow/BTLocalPaymentResult.swift:1]: I am not 100% sure when we need to import @BraintreeCore or not.

**Review by sshropshire — APPROVED** [review]

File: Sources/BraintreePaymentFlow/BTLocalPaymentResult.swift (hunk 2 above)
Comment (scannillo) [reply Sources/BraintreePaymentFlow/BTLocalPaymentResult.swift:1]: You import a module when you need to access it's classes & methods. In this case, `BTPostalAddress` is imported in parent class `BTPaymentFlowResult`.

The `canImport` syntax is for building our Demo app, and also for SPM _I believe_

**Review by KunJeongPark — APPROVED** [review]

**Pull request merged**